docs/USER_GUIDE.md

```diff
@@ -0,0 +1,543 @@
+- `rpc` generates RPC services (more information in the [RPC service part](#rpc-client))
+
+### Options
+
+The list of binapi-generator optional arguments in form `binapi-generator [OPTIONS] ARGS`
+
+- `binapi-generator -version` prints the generator version, for example, `govpp v0.8.0-dev`
+- `binapi-generator -gen=http` injects the `http` plugin to generate additional files.
+  More plugins are separated with comma, i.e. `-gen-http,rpc`. Note that `rpc` plugin is used by default
+- `binapi-generator -import-prefix=/desired/prefix/path` generates a custom prefix for imports in generated files
+- `binapi-generator -input-dir=/vpp/api/input/dir` sets the custom input directory instead of the default
+- `binapi-generator -output-dir=/bin/api/output/dir` sets the custom output directory
+- `binapi-generator -debug` prints some additional logs
+
+## VPP API calls
+
+GoVPP uses adapters to talk to VPP sockets.
+
+### Connection
+
+Two connection types to the binary API socket are exist - synchronous and asynchronous. GoVpp can connect to more VPPs
+by creating multiple connections (see the [multi-vpp example](../examples/multi-vpp/README.md))
+
+#### Synchronous connect
+
+The synchronous connecting creates a new adapter instance. The call blocks until the connection is established.
+
+```go
+conn, err := govpp.Connect(socketPath)
+if err != nil {
+// handle the error
+}
+defer conn.Disconnect()
+```
+
+#### Asynchronous connect
+
+The main difference between the synchronous and asynchronous connection is that the asynchronous connection does not
+block until the connection is ready. Instead, the caller receives a channel to watch connection events. Asynchronous
+connection defines additional parameters `attemptNum` and `interval` defining the number of reconnecting attempts, and
+an interval (in seconds) between those attempts.
+
+```go
+conn, connEv, err := govpp.AsyncConnect(socketPath, attemptNum, interval)
+if err != nil {
+// handle error
+}
+defer conn.Disconnect()
+
+// wait for the connection event
+e := <-connEv
+if e.State != core.Connected {
+// handle error in e.Error
+}
+```
+
+### Sending API messages
+
+Each binary API message in the Go-generated API is a data structure. The caller can send API messages either using
+the `Channel` (legacy method), or the `Stream client` (preferred). In-depth discussion about differences
+between `Channel` and `Stream` can be found [here](https://github.com/FDio/govpp/discussions/43).
+
+Messages can be requests or responses. The request might expect just one response (request), or more than one
+response (multirequest). It is possible to determine the message type out of its name.
+
+*_Note: the naming might not be consistent across the VPP API_*
+
+* *_Requests_* have no special suffix for the request, or `Dump` or `Get` for the multirequest.
+* *_Responses_* have a `Reply` suffix for the request or `Details` for multirequest.
+
+#### Channel
```

Comment:
As it's getting deprecated, maybe we could move this after the "Stream" section, and also add a Deprecated note with a ⚠️ emoji

Reply:
Or use

> **Warning**
> This is getting deprecated!